android/src/main/kotlin/io/piano/flutter/piano_analytics/PianoAnalyticsPlugin.kt

```diff
@@ -109,16 +139,92 @@ class PianoAnalyticsPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
             Event.Builder(
                 name = name,
                 properties = (it["data"] as? Map<*, *>)?.map { property ->
+                    val propertyName = property.key as? String
+                        ?: error("Undefined property name of event \"$name\"")
+                    val value = property.value as? Map<*, *>
+                        ?: error("Undefined property value of event \"$propertyName\"")
+                    val propertyValue = value["value"]
+                        ?: error("Undefined property value of event \"$propertyName\"")
+                    val forceType = (value["forceType"] as? String)?.let { type ->
+                        Property.Type.entries.firstOrNull { t -> t.prefix == type }
+                    }
                     getProperty(
-                        property.key as? String ?: error("Undefined property name of event \"$name\""),
-                        property.value
+                        propertyName,
+                        propertyValue,
+                        forceType
                     )
                 }?.toMutableSet() ?: mutableSetOf()
             ).build()
         }.toTypedArray()
         PianoAnalytics.getInstance().sendEvents(*events)
     }
 
+    private fun handleGetUser(result: Result) {
+        val user = PianoAnalytics.getInstance().userStorage.currentUser
+        if (user != null) {
+            result.success(
+                mapOf(
+                    "id" to user.id,
+                    "category" to user.category
+                )
+            )
+            return
+        }
+        result.success(null)
+    }
+
+    private fun handleSetUser(call: MethodCall) {
+        PianoAnalytics.getInstance().userStorage.currentUser = User(
+            call.arg("id"),
+            category = call.argument("category"),
+            shouldBeStored = call.argument<Boolean?>("enableStorage") ?: true
+        )
+    }
+
+    private fun handleGetVisitorId(result: Result) {
+        val pianoAnalytics = PianoAnalytics.getInstance()
+        result.success(pianoAnalytics.customVisitorId ?: pianoAnalytics.visitorId)
```

Comment:
There should be a check for visitorIdType not for null value in customVisitorId

Reply:
added check